### PR TITLE
selector: selection message and select all

### LINKED ISSUE #1834: Add drop-down selectors with multiple values
### Description

Hi,
When I use selectors, there isn't a combination that lets me have a multiple-value selector that is also a drop-down. 

### Solution Proposed

Something like drop-down and check boxes for example. The following doesn't give an error but also doesn't "work":

```
tgb.selector
    (...
        multiple=True,
        dropdown=True,
        mode="check",)
```

**Another thing that could be great, is if the callback is only called after I select/unselect all my values (when I go out of the drop-down list). Right now, every time I check a box, it calls on_action, so if I check (or uncheck) a lot of boxes, it calls the Callback many times.**

### Impact of Solution

To give an example, I used a side pane to hold a long list of checkboxes, but it would have been quicker to have this drop-down list, and also more efficient. 

### Additional Context

_No response_

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1834

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- selection message property dynamic(string)
- show_select_all property boolean

## Related Tickets & Documents

- Closes #1834 

## How to reproduce the issue

```py
from taipy.gui import Gui, State

sel = None
sel_msg = None

def on_change(state: State, var_name: str, value):
  if var_name == "sel":
    state.sel_msg = f"{len(value)} selected elements"

page = """
<|{sel}|selector|dropdown|multiple|lov=Item 1;Item2 2;Item 3;Item 4|selection_message={sel_msg}|show_select_all|label=select an item|>
<|{sel}|selector|multiple|lov=Item 1;Item2 2;Item 3;Item 4|show_select_all|>
"""

if __name__ == "__main__":
  Gui(page).run(title="1834 Selector Dropdown Multiple Message")

```

### CODE REVIEW

**Review by namnguyen20999 — APPROVED** [review]
LGTM

File: taipy/gui/_renderers/factory.py [hunk 1]
```diff
@@ -507,6 +506,8 @@ class _Factory:
                 ("label",),
                 ("mode",),
                 ("lov", PropertyType.lov),
+                ("selection_message", PropertyType.dynamic_string),
```
Comment (dinhlongviolin1) [inline taipy/gui/_renderers/factory.py:509]: I think `selected_message` sounds better

**Review by dinhlongviolin1 — APPROVED** [review]
Lgtm! just a naming suggestion

**Review by FabienLelaquais — APPROVED** [review]
👌

File: taipy/gui/_renderers/factory.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/_renderers/factory.py:509]: I think `selection_message` is closer to what this actually does

**Comment by FlorianJacta** [comment]
This is not working with filter=True. I'll create the issue on that

**Pull request merged**